fs/du.go

```diff
@@ -0,0 +1,43 @@
+package fs
+
+import (
+	"os"
+	"path/filepath"
+	"syscall"
+)
+
+type Usage struct {
+	Inodes int64
+	Size   int64
+}
+
+// DiskUsage counts the number of inodes and disk usage for the resources under
+// path.
+func DiskUsage(root string) (Usage, error) {
+	type inode struct {
+		// TODO(stevvooe): Can probably reduce memory usage by not tracking
+		// device, but we can leave this right for now.
+		dev, ino uint64
+	}
+
+	var (
+		size   int64
+		inodes = map[inode]struct{}{} // expensive!
+	)
+
+	err := filepath.Walk(root, func(path string, fi os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		stat := fi.Sys().(*syscall.Stat_t)
+		inodes[inode{dev: stat.Dev, ino: stat.Ino}] = struct{}{}
+		size += fi.Size()
```

Comment:
size should only be added if the inode didn't already exist in the map, could possibly avoid the check if the number of links is 1.